iceoryx2/src/service/mod.rs

```diff
@@ -272,6 +293,23 @@ pub struct ServiceDetails<S: Service> {
     pub dynamic_details: Option<ServiceDynamicDetails<S>>,
 }
 
+#[allow(missing_docs)]
+#[derive(serde::Serialize)]
+pub struct SerializableServiceDetails {
```

Comment:
Could you please move this as well into a separate file `serializable_service_details.rs`